src/webui/api/apicontroller.h

```diff
@@ -67,6 +74,9 @@ class APIController : public QObject
     void setResult(const QJsonObject &result);
 
 private:
+    virtual bool isActionSafe(const QString &actionName) const = 0;
```

Comment:
I'm thinking of another way to realize it:
1. This function definition should stay in `APIController` class (in .cpp file!), it doesn't make sense to have every derived class to impl the same thing.
2. This (base) class should have `const QSet<QString> m_safeActions, m_unsafeActions;` variables. And let derived classes provide the values on construction.

IMO this would lead to simpler code and easier maintenance.

Reply:
This is definitely simpler and makes more sense 👍 Will implement.